frontend/src/pages/WaitingPage.tsx

```diff
@@ -4,12 +4,15 @@ import { CircularProgress, Typography } from "@mui/material";
 import { Stack } from "@mui/system";
 
 import { useSocket } from "src/contexts/WsContext";
+import { useGetUsername } from "src/hooks/useUsers";
 
 import { Match } from "~shared/types/api/match.dto";
 
 export const WaitingPage = () => {
   const navigate = useNavigate();
   const { socket, connect } = useSocket();
+  const { getUsername } = useGetUsername();
+
```

Comment:
And we can use a `useEffect` hook to set the message when all the data is available.
```
  useEffect(() => {
    if (roomId && userNameOne && userNameTwo) {
      setMessage(
        `Found a match between ${userNameOne.name} and ${userNameTwo.name}!
        Room ID: ${roomId}
        Loading...`,
      );
    }
  }, [roomId, userNameOne, userNameTwo]);
```